### ISSUE TITLE
TZP overview

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [arkenfox/TZP2#8]
TBA

**Comment by Thorin-Oakenpants** [comment]
rambling NOTES:

display data
* some data displayed is not in the FP, they are cosmetic, because they are already in the FP elsewhere: e.g. wasm enabled is present is window properties, but we want to be able to notate the slider info. Or some are legacy, such as performance.marks, which are relevant for TB < 115

data
* we record errors, alerts, lies, bypasses, methods, etc. The ability exists to add these per section (i.e sData + gData)  - so far I have only hooked up errors. This gets slightly complicated due to the need to cache some of them

data
* the original TZP used flat arrays and strings e.g. ["metric: value", "metric: hash"] and TZP2 doesn't. So you can see errors is sectionalized in gData in an object, but the others aren't. So there's a bit of hybrid code going on, as I have refactored/rebuilt things in place, rather than rebuild entirely from scratch. And some metrics aren't hooked up to the new data methods, and some require a rewrite to use new logic and be consistent

runs
* we run automatically on page load - in future we might not do this and require a user action first as this allows collecting some extra metrics
* we also currently allow a global rerun, and individual section reruns. I built this functionality in so I could test what happens when you do pref flips without reloading, etc. It's also useful for running test simulations via code: e.g. cycling thru a set of results to make sure my metrics and displays and notations match up.
* the screen section on desktop updates the values (but not section hashes etc) in real time on desktop on resize events
* for simplicity (and stability), with future work we could remove this and only enable it for say local:host and file:// schemas .. or do away with it. But for now it stays

TZP2
*  one major change is how I treat different scenarios
   - e.g instead of methods, I will try to put that into the actual metric
      - e.g. if canvas is randomized, it is either `randomized per execution` or `randomized` (stealth) - there's more to fingerprinting protections in canvas than just this
   - e.g if an extension used to return a unexpected typeof, I would report `blocked`, and color it red as a lie, etc
      - TZP2 is simpler, it just returns a TypeError (recorded in the errors fingerprint)
   - e.g. if an extension causes paradoxical results (due to equivalency checks)
* another is how I treat notations. Sometimes I report a red fail if there is a proxy lie or css psuedo lie check, other times I don't, so this all needs tightening up
* another is some lies I silently bypass (from equivalency), other times not
* so long story short, it's all a bit inconsistent
